### ISSUE TITLE
filter.REPLY not working [BUG]

### ISSUE THREAD

**Issue by cciliber** [python-telegram-bot/python-telegram-bot#3621]
### Steps to Reproduce

1. send a reply to the bot via telegram
2. try to catch the reply via `MessageHandler(filters.REPLY, reply_handler)` for some `reply_handler` of your choice. For instance

```python
async def reply_with_echo(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await context.bot.send_message(chat_id=update.effective_chat.id, text="To message '" + update.message.reply_to_message.text +"' you said: " + update.message.text)


if __name__ == '__main__':
    application = ApplicationBuilder().token(TOKEN).build()

    reply_handler = MessageHandler(filters.TEXT & filters.REPLY, reply_with_echo)
    application.add_handler(reply_handler)

    application.run_polling()

```

### Expected behaviour

a message from the bor

### Actual behaviour

no message from the bot

### Operating System

macOS 13

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot 20.1
Bot API 6.5
Python 3.11.0 | packaged by conda-forge | (main, Jan 15 2023, 05:44:48) [Clang 14.0.6 ]
```

### Additional Context

It's only `filter.REPLY` that doesn't seem to work, but the reply message is captured in the message (maybe I am using it incorrectly? But I wasn't able to get clear guidance from the docs). 

In other words, if I write something like 

```python
async def reply_with_echo(update: Update, context: ContextTypes.DEFAULT_TYPE):
    print(update.message)
    if update.message.reply_to_message:
        await context.bot.send_message(chat_id=chat_id(update), text="To message '" + update.message.reply_to_message.text +"' you said: " + update.message.text)
    else:
        await context.bot.send_message(chat_id=chat_id(update), text=update.message.text)
```

And then catch it without `filter.REPLY`, namely 

```python

if __name__ == '__main__':
    application = ApplicationBuilder().token(TOKEN).build()

    reply_handler = MessageHandler(filters.TEXT, reply_with_echo)
    application.add_handler(reply_handler)

    application.run_polling()

```

Then everything works as desired.

**Comment by clot27** [comment]
The function `reply_with_echo()` you provided works perfectly fine for me with `filters.TEXT` and `filters.REPLY`..?

**Comment by cciliber** [comment]
Ok, got it. In the original code I had, I had another `MessageHandler` that was consuming `filters.TEXT` without `filters.REPLY` and it was added to the application before the `reply_handler` and ended up consuming all text messages before they could get to the latter. 

I should have tested my MWE rather than just deleting the lines that I thought were only not strictly necessary to reproduce the error. Apologies.
